packages/compiler-core/src/utils.ts

```diff
@@ -60,7 +60,7 @@ export function isCoreComponent(tag: string): symbol | void {
   }
 }
 
-const nonIdentifierRE = /^\d|[^\$\w]/
+const nonIdentifierRE = /^\d|[^\$\w\u4e00-\u9fff]/
```

Comment:
I believe we should support all valid identifier characters instead of merely Chinese.